include/oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h

```diff
@@ -106,27 +171,45 @@ class sycl_backend
                       report_value_v<SelectionHandle, execution_info::task_time_t>)
         {
             std::chrono::steady_clock::time_point t0;
+            bool use_event_profiling = q.template has_property<sycl::property::queue::enable_profiling>();
             if constexpr (report_value_v<SelectionHandle, execution_info::task_time_t>)
             {
-                t0 = std::chrono::steady_clock::now();
+                if (!use_event_profiling)
+                {
+                    t0 = std::chrono::steady_clock::now();
+                }
             }
             auto e1 = f(q, std::forward<Args>(args)...);
-            auto e2 = q.submit([=](sycl::handler& h) {
-                h.depends_on(e1);
-                h.host_task([=]() {
-                    if constexpr (report_value_v<SelectionHandle, execution_info::task_time_t>)
-                        s.report(execution_info::task_time, (std::chrono::steady_clock::now() - t0).count());
-                    if constexpr (report_info_v<SelectionHandle, execution_info::task_completion_t>)
-                    {
+            if constexpr(report_info_v<SelectionHandle, execution_info::task_completion_t>){
+                auto e2 = q.submit([=](sycl::handler& h){
+                    h.depends_on(e1);
+                    h.host_task([=](){
                         s.report(execution_info::task_completion);
-                    }
+                    });
                 });
-            });
-            return async_waiter{e2};
+                return async_waiter{e2, std::make_shared<SelectionHandle>(s)};
+            }
+            else if constexpr(report_value_v<SelectionHandle, execution_info::task_time_t>){
```

Comment:
Why `else if` here? Cannot both task time and task completion be reported?

Reply:
There shouldn't be this restriction. I think it should an if and not an if-else.

Reply:
Fixed. Changed to "if"

Reply:
This is insufficient; more details in another comment.